plugins/faustwp/includes/replacement/functions.php

```diff
@@ -131,3 +131,106 @@ function is_wp_link_ajax_request(): bool {
 		&& ! empty( $_POST['action'] )
 		&& 'wp-link-ajax' === $_POST['action'] );
 }
+
+/**
+ * Get all site URLs for each possible HTTP protocol
+ */
+function faustwp_get_wp_site_urls() {
+
+	$site_url = site_url();
+	$host_url = wp_parse_url( $site_url, PHP_URL_HOST );
+
+	if ( ! is_string( $host_url ) ) {
+		return apply_filters( 'faustwp_get_wp_site_urls', array( $site_url ) );
+	}
+
+	$is_https = substr( $site_url, 0, 6 ) === 'https:';
+
+	return apply_filters(
+		'faustwp_get_wp_site_urls',
+		array(
+			$is_https ? "https://$host_url" : "http://$host_url",
+			$is_https ? "http://$host_url" : "https://$host_url",
+			"//$host_url",
+		)
+	);
+}
+
+/**
+ * Get all media urls based off the available site urls
+ *
+ * @return array
+ */
+function faustwp_get_wp_media_urls() {
+	$site_urls  = faustwp_get_wp_site_urls();
+	$upload_url = faustwp_get_relative_upload_url( $site_urls );
+
+	if ( ! is_string( $upload_url ) ) {
+		return apply_filters( 'faustwp_get_wp_site_media_urls', array() );
+	}
+
+	$media_urls = array();
+	foreach ( $site_urls as $site_url ) {
+		$media_urls[] = $site_url . $upload_url;
+	}
+
+	return apply_filters( 'faustwp_get_wp_site_media_urls', $media_urls );
+}
+
+
+/**
+ * Gets the relative wp-content upload URL.
+ *
+ * @param array<string> $site_urls An array of site URLs.
+ *
+ * @return string The relative upload URL.
+ */
+function faustwp_get_relative_upload_url( $site_urls ) {
+	$upload_dir = wp_upload_dir()['baseurl'];
+
+	foreach ( $site_urls as $site_url ) {
+		if ( false !== strpos( $upload_dir, $site_url ) ) {
+			return (string) str_replace( $site_url, '', $upload_dir );
+		}
+	}
+
+	return '';
+}
+
+/***
+ * Replaces the media URL for various media urls
+ *
+ * @param string $content The content to be updated with the new media URL.
+ * @param array  $wp_media_urls An array of media URLS.
+ * @param string $replace_url The media URL to be updated to.
+ *
+ * @return string The replaced string
+ */
+function faustwp_replace_media_url( string $content, array $wp_media_urls, string $replace_url ) {
```

Comment:
I am not sure if this function is needed. It looks like a wrapper to the `str_replace` with an alternative parameters order.
Please remember that the `str_replace` can also accept the array as a search param, so no need to loop through each one.

Reply:
We use this function in 2 places.Happy to extract the functionality out if you prefer?

Reply:
Yeah, my point is that the function itself is just a wrapper for the `str_replace`. We don't actually do anything except calling the `str_replace` here.

Reply:
Ah sorry I see your point now. I will remove it as a function should really be doing more that one operation